### PR TITLE
feat(schematics): create ng add schematics

### PR DESCRIPTION
**Pull request by thekiba**
Issue #53

carbon 10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "contributors:add": "npx all-contributors add",
     "contributors:generate": "npx all-contributors generate",
     "test:ci": "npx cross-env NODE_ENV=build npm run test && npm run test:jest",
-    "build:lib": "npx ng build spectator"
+    "build:lib": "npx ng build spectator && npm run build:schematics",
+    "build:schematics": "tsc -p projects/spectator/schematics/tsconfig.json && cp projects/spectator/schematics/src/collection.json dist/spectator/schematics/ && cp projects/spectator/schematics/src/ng-add/schema.json dist/spectator/schematics/ng-add/"
```
Comment (thekiba) [inline package.json]: I do not know how to implement building and copying schematics files natively for angular.json. So I just created it as is for to find solution together.

Comment (NetanelBasal) [reply package.json]: Check out https://github.com/angular/angular-cli/blob/master/docs/documentation/stories/asset-configuration.md#writing-assets-outside-of-dist.

**Comment by NetanelBasal** [comment]
This is super awesome, thanks!

File: package.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
     "contributors:add": "npx all-contributors add",
```
Comment (NetanelBasal) [inline package.json]: Add yourself to the contributors' list; You can find the scripts in this file.

File: package.json (hunk 1 above)
Comment (thekiba) [reply package.json]: I thought about it too, but the ng-packagr builder does not have the `assets` config in angular.json

Comment (NetanelBasal) [reply package.json]: So check out Akita's version. https://github.com/datorama/akita/blob/master/package.json#L17

Comment (thekiba) [reply package.json]: Just copy all of source files (ts+js)? I will make commit with it soon.

**Comment by NetanelBasal** [comment]
I'm wondering what happens to the tests? It seems like they don't run.

**Comment by thekiba** [comment]
I tried to run `npm run test:ci` on the `master` branch, and they also failed.

```
SUMMARY:
✔ 65 tests completed
✖ 2 tests failed
```

**Comment by NetanelBasal** [comment]
✖ 2 tests failed? what is the output?

**Comment by thekiba** [comment]
```
FAILED TESTS:
  DomSelectorsComponent
    ✖ should allow querying by text
      HeadlessChrome 0.0.0 (Mac OS X 10.14.3)
    TypeError: node.matches is not a function
    ✖ should allow querying by label
      HeadlessChrome 0.0.0 (Mac OS X 10.14.3)
    TypeError: node.matches is not a function
```

**Comment by NetanelBasal** [comment]
I need to investigate it. I'll do it later. When I resolve this, I will merge.

**Comment by NetanelBasal** [comment]
Pull from `master` and it should work.

**Comment by thekiba** [comment]
When i use yarn — succeed, and when npm — failed.
What are you using? And what is it versions?

**Comment by thekiba** [comment]
It's not fixes, just added missing dependency 🐒

**Pull request merged**